code/repoutil

```diff
@@ -560,12 +560,17 @@ def copy_branches(source_branch, dest_branch):
     reposadocommon.writeAllBranchCatalogs()
 
 
-def delete_branch(branchname):
+def delete_branch(branchname, force=False):
     '''Deletes a branch'''
     catalog_branches = reposadocommon.getCatalogBranches()
     if not branchname in catalog_branches:
         reposadocommon.print_stderr('Branch %s does not exist!', branchname)
         return
+    if force:
+        del catalog_branches[branchname]
+        deleteBranchCatalogs(branchname)
+        reposadocommon.writeCatalogBranches(catalog_branches)
+        return
```

Comment:
Could we instead construct this like it is in copy_branch so we have less code duplication?